test/sanitytest.c

```diff
@@ -49,6 +49,15 @@ static int test_sanity_enum_size(void)
     return 1;
 }
 
+static int test_sanity_int_size(void)
+{
+    if (!TEST_size_t_eq(sizeof(int), 4)
+        || !TEST_int_eq(CHAR_BIT, 8)
+        || !TEST_size_t_lt(sizeof(int), sizeof(int64_t)))
```

Comment:
That's a false assumption.  By definition, an `int` is at least 16 bits wide and at most as wide as a `long`.  If you want exact width, you have to got with the `intN_t` types.
(if nothing else, the story of `crypto/asn1/x_long.c` proves the short sightedness of assuming that a type with inexact width will stay the same forever).

Either way, having all three tests is redundant.  Just as an example, if the first two are true, then the last one *must* be logically true.